eng/restore/dirs.proj

```diff
@@ -1,5 +1,5 @@
-<Project InitialTargets="IncludeProjectReferences">
```

Comment:
This change is unrelated to the PR, please factor it out.

Reply:
This is required for this pr because the restore started by the dirs.proj is enforcing an additional buildconfiguration corresponding to targetframework netcoreapp5.0 to netcoreapp.deproj. We dont want to restore this proj for netcoreapp5.0

Reply:
Indeed.  Without this change dirs.proj ends up restoring all the external projects for "netcoreapp5.0" (because it calls SDK's restore target and leaks the Configuration global property).  This is what we were discussing in standup and in the other PR where you moved this.

Reply:
If this is blocking then I'm fine with the change and I'll explore later for other options.